node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -159,6 +161,24 @@ impl Queues {
 		self.pop_best_effort().map(|d| d.1)
 	}
 
+	/// Reprioritizes any participation requests pertaining to the
+	/// passed candidates from best effort to priority.
+	pub async fn prioritize_if_present(
+		&mut self, 
+		sender: &mut impl overseer::DisputeCoordinatorSenderTrait,
+		receipt: &CandidateReceipt,
+	) -> std::result::Result<(), QueueError>{
+		if self.priority.len() >= PRIORITY_QUEUE_SIZE {
+			return Err(QueueError::PriorityFull)
+		}
+
+		let comparator = CandidateComparator::new(sender, receipt).await.map_err(|_e| QueueError::CouldNotGenerateComparator)?;
```

Comment:
Consistency. [Here](https://github.com/paritytech/polkadot/blob/16edc2b95dce8ce1e09a588b550c0cbcd5b66120/node/core/dispute-coordinator/src/participation/queues/mod.rs#L149) we just bubble up the error as is.